crypto/dilithium/ml_dsa.h

```diff
@@ -9,15 +9,54 @@
 #include <openssl/base.h>
 #include <openssl/evp.h>
 
+#define MLDSA44_PUBLIC_KEY_BYTES 1312
+#define MLDSA44_PRIVATE_KEY_BYTES 2560
```

Comment:
In [draft-ietf-lamps-dilithium-certificates](https://datatracker.ietf.org/doc/draft-ietf-lamps-dilithium-certificates/) Appendix B, it lists the private key size as 32-bytes whereas this 2560 value aligns with FIPS-204. Can you explain the disconnect for me?

Reply:
Check https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.204.pdf page 16 for the parameter sets that match these parameters. In [draft-ietf-lamps-dilithium-certificates](https://datatracker.ietf.org/doc/draft-ietf-lamps-dilithium-certificates/) we allow the private key to be stored as a seed.

See various discussions here: https://groups.google.com/a/list.nist.gov/g/pqc-forum/c/949fsZRC6Ik https://groups.google.com/a/list.nist.gov/g/pqc-forum/c/2EWXvBaE-Yw/